mlflow/server/handlers.py

```diff
@@ -1103,6 +1104,134 @@ def _default_history_bulk_impl():
     }
 
 
+@catch_mlflow_exception
+@_disable_if_artifacts_only
+def get_metric_history_bulk_interval_handler():
+    MAX_RUNS_GET_METRIC_HISTORY_BULK = 100
+    MAX_RESULTS_PER_RUN = 2500
+    MAX_RESULTS_GET_METRIC_HISTORY = 25000
+
+    args = request.args
+    run_ids = args.to_dict(flat=False).get("run_ids", [])
+    if not run_ids:
+        raise MlflowException.invalid_parameter_value(
+            "GetMetricHistoryBulkInterval request must specify at least one run_id."
+        )
+    if len(run_ids) > MAX_RUNS_GET_METRIC_HISTORY_BULK:
+        raise MlflowException.invalid_parameter_value(
+            f"GetMetricHistoryBulkInterval request must specify at most "
+            f"{MAX_RUNS_GET_METRIC_HISTORY_BULK} run_ids. Received {len(run_ids)} run_ids."
+        )
+    if max_results := args.get("max_results"):
+        max_results = int(max_results)
```

Comment:
do we need this casting?

Reply:
Yes, otherwise it's string

Reply:
Can we use `_get_request_message` to validate and cast parameters using the proto definition?

Reply:
Seems _get_request_message doesn't really check schema for GET requests with query_strings.. https://github.com/mlflow/mlflow/blob/a81ab9d461c7242a92209c4a1e5e69b15e2c6671/mlflow/server/handlers.py#L412-L415 Do we want to fix it in this PR? BTW this is not a public API, I feel like we don't even need to add proto 🤔

Reply:
We should use proto when we add a new endpont. It doesn't matter whether it's public or private.